src/deadline/job_attachments/api/manifest.py

```diff
@@ -247,3 +260,191 @@ def process_output(status: FileStatus, path: str, output_diff: ManifestDiff):
             process_output(fast_diff_item[1], fast_diff_item[0], output)
 
     return output
+
+
+def _manifest_upload(
+    manifest_file: str,
+    s3_bucket_name: str,
+    s3_cas_prefix: str,
+    boto_session: boto3.Session,
+    s3_key_prefix: Optional[str] = None,
+    logger: ClickLogger = ClickLogger(False),
+):
+    """
+    BETA API - This API is still evolving but will be made public in the near future.
+    API to upload a job attachment manifest to the Content Addressable Storage. Manifests will be
+    uploaded to s3://{s3_bucket_name}/{cas_prefix}/Manifests/{s3_key_prefix}/{manifest_file_name} as per the Deadline CAS folder structure.
+    manifest_file: File Path to the manifest file for upload.
+    s3_bucket_name: S3 bucket name.
+    boto_session: S3 Content Addressable Storage prefix.
+    s3_key_prefix: [Optional] S3 prefix path to the Content Addressable Storge.
+    boto_session: Boto3 session.
+    logger: Click Logger instance to print to CLI as test or JSON.
+    """
+    # S3 metadata
+
+    # Upload settings:
+    s3_metadata: Dict[str, Any] = {"Metadata": {}}
+    s3_metadata["Metadata"]["file-system-location-name"] = manifest_file
+
+    # Always upload the manifest file to case root /Manifest with the original file name.
+    manifest_path: str = "/".join(
+        [s3_cas_prefix, S3_MANIFEST_FOLDER_NAME, s3_key_prefix, Path(manifest_file).name]
+        if s3_key_prefix
+        else [s3_cas_prefix, S3_MANIFEST_FOLDER_NAME, Path(manifest_file).name]
+    )
+
+    # S3 uploader.
+    upload = S3AssetUploader(session=boto_session)
+    with open(manifest_file) as manifest:
```

Comment:
This API doesn't have the path check like in the CLI, is this intended?

Reply:
Yeah, let it fail with normal file errors, this is an API afterall. Why wrap errors when the `open` error is very specific of what is wrong. I don't like wrapping errors :)